### PR TITLE
chore: rename tokenized prb flag to ece

### LINKED ISSUE #9720: Rename `_wcpay_feature_tokenized_cart_prb` Feature Flag to `_wcpay_feature_tokenized_cart_ece`
To unify the tokenized cart approach, we’ll rename the `_wcpay_feature_tokenized_cart_prb` feature flag to `_wcpay_feature_tokenized_cart_ece`. This aligns with the new structure in which the ECE and Tokenized Cart implementations will share common ground, with the Store API as the point of interface.

For background: pbIwZe-2ur-p2

## Acceptance Criteria:

- `_wcpay_feature_tokenized_cart_prb` flag is renamed to `_wcpay_feature_tokenized_cart_ece` throughout the codebase.
- Verify no functionality is impacted by the renaming.
- Confirm that tests relying on the original feature flag are updated accordingly.
- All related documentation, comments, and references are updated to reflect the new flag name

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/9720

#### Changes proposed in this Pull Request

Part of pbIwZe-2ur-p2 - renaming the "Tokenized Cart PRB" to "Tokenized Cart ECE" feature flag.

The purpose of this PR is iterative, just to rename the flag, without caring for the actual feature.
The work to convert the feature to use the ECE will come later.

The dev tools will also be updated.

#### Testing instructions

No change in functionality - this is just renaming a feature flag for a feature that is not live.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/checkout/blocks/index.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ if ( getUPEConfig( 'isWooPayEnabled' ) ) {
 }
 
 if ( getUPEConfig( 'isPaymentRequestEnabled' ) ) {
-	if ( getUPEConfig( 'isTokenizedCartPrbEnabled' ) ) {
+	if ( getUPEConfig( 'isTokenizedCartEceEnabled' ) ) {
 		registerExpressPaymentMethod(
 			tokenizedCartPaymentRequestPaymentMethod( api )
```
Comment (frosso) [inline client/checkout/blocks/index.js:165]: I decided not to rename this (and the `tokenized-payment-request` directory). I think it's ok to have that work to be separate 👍

File: includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php [hunk 2]
```diff
@@ -116,10 +116,9 @@ public function init() {
 	 * @return void
 	 */
 	public function display_express_checkout_separator_if_necessary( $separator_starts_hidden = false ) {
-		$html_id = WC_Payments_Features::is_tokenized_cart_prb_enabled() ? 'wcpay-payment-request-button-separator' : 'wcpay-express-checkout-button-separator';
```
Comment (frosso) [inline includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php:119]: No need for a differentiation, at this time.

File: includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php [hunk 3]
```diff
@@ -145,7 +144,7 @@ public function display_express_checkout_buttons() {
 				$this->platform_checkout_button_handler->display_woopay_button_html();
 			}
 
-			if ( WC_Payments_Features::is_tokenized_cart_prb_enabled() ) {
+			if ( WC_Payments_Features::is_tokenized_cart_ece_enabled() ) {
 				$this->payment_request_button_handler->display_payment_request_button_html();
```
Comment (frosso) [inline includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php:148]: This (and its class) will also iteratively be renamed

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9715` or branch name `chore/rename-tokenized-prb-flag-to-tokenized-ece` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=chore/rename-tokenized-prb-flag-to-tokenized-ece) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: a71e88e435d7372cf8dbcd68442791b39de4951e
- Build time: 2024-11-14 17:54:35 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by FangedParakeet — APPROVED** [review]
LGTM. 👍 

I did consider possibly disambiguating this feature from the ECE as well as the PRB--perhaps naming it something like `_wcpay_feature_tokenized_cart_ece`/`#wcpay-express-payment-button-separator`. Though, this feature will be relatively short-lived and definitely won't outlive the ECE. Moreover, this implementation _will_ be somewhat tightly tied to the ECE's and presumably we would need a new feature flag for another future EPM integration besides. 

Anyways, lots of chatter to basically say: Saul Goodman. :shipit:

**Pull request merged**